Vagrantfile

```diff
@@ -6,6 +6,196 @@ $NODEMEM=256
 # Overwrite host locale in ssh session
 ENV["LC_ALL"] = "en_US.UTF-8"
 
+BOOTSTRAP_SCRIPT=<<__EOF__
+#!/bin/bash
+# Bootstrap the Ansible Lightbulb workshop environment on a Vagrant box
+
+#
+# Bash shell settings: exit on failing commands, unbound variables
+#
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#
+# Variables
+#
+# Color definitions
+readonly reset='\e[0m'
+readonly red='\e[0;31m'
+readonly yellow='\e[0;33m'
+readonly cyan='\e[0;36m'
+
+readonly workshop_user=vagrant
+
+readonly workshop_key="/home/${workshop_user}/.ssh/id_rsa"
+readonly workshop_inventory="/home/${workshop_user}/hosts.ini"
+readonly workshop_ansible_config="/home/${workshop_user}/.ansible.cfg"
+
+readonly workshop_git_repo="https://github.com/ansible/lightbulb.git"
+
+#
+# Functions
+#
+
+main() {
+  info "Bootstrapping Ansible on host ${HOSTNAME}."
+
+  ensure_ansible_installed
+  install_vagrant_private_key
+  configure_ansible
+  install_inventory
+  install_lightbulb
+
+}
+
+# Ansible installation
+ensure_ansible_installed() {
+  if ! is_ansible_installed; then
+    distro=$(get_linux_distribution)
+    "install_ansible_${distro}" || die "Distribution ${distro} is not supported"
+  fi
+
+  info "Ansible version"
+  ansible --version
+}
+
+is_ansible_installed() {
+  which ansible-playbook > /dev/null 2>&1
+}
+
+# Print the Linux distribution
+get_linux_distribution() {
+
+  if [ -f '/etc/redhat-release' ]; then
+
+    # RedHat-based distributions
+    cut --fields=1 --delimiter=' ' '/etc/redhat-release' \
+      | tr "[:upper:]" "[:lower:]"
+
+  elif [ -f '/etc/lsb-release' ]; then
+
+    # Debian-based distributions
+    grep DISTRIB_ID '/etc/lsb-release' \
+      | cut --delimiter='=' --fields=2 \
+      | tr "[:upper:]" "[:lower:]"
+
+  fi
+}
+
+# Install Ansible on a Fedora system.
+# The version in the repos is fairly current, so we'll install that
+install_ansible_fedora() {
+  info "Fedora: installing Ansible from distribution repositories"
+  dnf -y install git ansible
+}
+
+# Install Ansible on a CentOS system from EPEL
+install_ansible_centos() {
+  info "CentOS: installing Ansible from the EPEL repository"
+  yum -y install epel-release
+  yum -y install git ansible
```

Comment:
The first Ansible Essentials workshop assignment is to install `ansible` itself -- by putting this function in we are making that pointless. I want users of this content to experience install ansible and how easy it is.

Reply:
That's fine, I agree.